### PR TITLE
Improve docs

### PR DESCRIPTION
**Pull request by dstepanov**
My commit didn't got pushed and merged

### CODE REVIEW

File: src/main/docs/guide/whatsNew.adoc [hunk 1]
```diff
@@ -1,5 +1,8 @@
 === Micronaut Data 4.0
-- Hibernate6 Support
+- Hibernate 6
+- Hibernate Reactive 6
```
[the remark this answers is not in the document]
Comment (dstepanov) [reply src/main/docs/guide/whatsNew.adoc]: Corrected

File: src/main/docs/guide/whatsNew.adoc [hunk 2]
```diff
@@ -1,5 +1,8 @@
 === Micronaut Data 4.0
-- Hibernate6 Support
+- Hibernate 6
+- Hibernate Reactive 2 (Hibernate 6 compatible)
```
Comment (sdelamo) [inline src/main/docs/guide/whatsNew.adoc:3]:
```suggestion
- https://hibernate.org/reactive/releases/2.0/[Hibernate Reactive 2] (Hibernate 6 compatible)
```

File: src/main/docs/guide/whatsNew.adoc [hunk 3]
```diff
@@ -1,5 +1,8 @@
 === Micronaut Data 4.0
-- Hibernate6 Support
+- Hibernate 6
```
Comment (sdelamo) [inline src/main/docs/guide/whatsNew.adoc:2]:
```suggestion
- https://hibernate.org/orm/documentation/6.0/[Hibernate 6]
```

File: src/main/docs/guide/breaks.adoc [hunk 4]
```diff
@@ -14,6 +14,10 @@ public interface BookRepository implements CrudRepository<@jakarta.validation.Va
 }
 ----
 
+===== Repositories now return `List`
+
+Find all return type changed to be `List`
```
Comment (sdelamo) [inline src/main/docs/guide/breaks.adoc:19]:
```suggestion
Find all return type changed to be `List` instead of `Iterable`
```

**Review by sdelamo — CHANGES_REQUESTED** [review]

**Pull request merged**